binlog.cpp

```diff
@@ -154,16 +156,22 @@ void BinLog::_send_ack(uint32_t seqno)
 
 void BinLog::_logging_data_process(mavlink_remote_log_data_block_t *msg)
 {
-    ssize_t r;
-
-    if (lseek(_file, msg->seqno * MAVLINK_MSG_REMOTE_LOG_DATA_BLOCK_FIELD_DATA_LEN, SEEK_SET) < 0) {
-        log_error_errno(errno, "lseek failed (%m)");
-        return;
-    }
-
-    r = write(_file, msg->data, MAVLINK_MSG_REMOTE_LOG_DATA_BLOCK_FIELD_DATA_LEN);
-    if (r != MAVLINK_MSG_REMOTE_LOG_DATA_BLOCK_FIELD_DATA_LEN) {
-        log_error_errno(errno, "Error writing data (%m)");
+    int result;
+
+    result = _ringbuffer.add(&msg->data, msg->seqno);
+    switch (result) {
+    case _ringbuffer.overflow:
+        // This is kinda desperate, because when ringbuffer filled, we tried to write
+        // TODO control these writes - maybe stop logging and starting again?
+        log_warning("Buffer overflow, discarding message, writing");
+        _ringbuffer.write_partial(_file);
+        break;
+    case _ringbuffer.full:
+        log_warning("Buffer full, writing");
```

Comment:
This is a normal case right? So the log level should be info or debug otherwise it will print a lot of information that is not necessary.